src/util/buf.rs

```diff
@@ -219,14 +218,20 @@ impl PipeBuffer {
         let mut readed;
         {
             let (left, right) = self.slice_append();
-            match try!(r.try_read(left)) {
-                None => return Ok(0),
-                Some(l) => readed = l,
+            match r.read(left) {
+                Ok(l) => readed = l,
+                Err(e) => {
+                    return if e.kind() == ErrorKind::WouldBlock {
```

Comment:
can we construct a Reader/Writer which can return WouldBlock to cover the test now?